specs/Workers.md

```diff
@@ -0,0 +1,1053 @@
+               if(workersCollection)
+               {
+                  UINT32 workersCount = 0;
+                  CHECK_FAILURE(workersCollection->get_Count(&workersCount));
+
+                  std::wstringstream message{};
+                  message << L"No of shared workers created: " << workersCount << std::endl;
+
+                  for (UINT32 i = 0; i < workersCount; i++)
+                  {
+                      Microsoft::WRL::ComPtr<ICoreWebView2SharedWorker> sharedWorker;
+                      CHECK_FAILURE(workersCollection->GetValueAtIndex(i, &sharedWorker));
+
+                      wil::com_ptr<ICoreWebView2Worker> worker;
+                      if (SUCCEEDED(sharedWorker->QueryInterface(IID_PPV_ARGS(&worker))))
+                      {
+                          wil::unique_cotaskmem_string scriptUrl;
+                          CHECK_FAILURE(worker->get_ScriptUrl(&scriptUrl));
+
+                          message << L"ScriptUrl: " << scriptUrl.get();
+                          message << std::endl;
+                      }
+                  }
+
+                  m_appWindow->AsyncMessageBox(
+                      std::move(message.str()), L"Get all shared workers");
+               }
+                return S_OK;
+            })
+            .Get()));
+  }
+```
+
+# API Details
+```
+/// State of the service worker.
+[v1_enum]
+typedef enum COREWEBVIEW2_SERVICE_WORKER_STATE {
+  /// The service worker is new and has not been installed yet.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_NEW,
+  /// The service worker is installing.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_INSTALLING,
+  /// The service worker is installed. The service worker in this state is considered a waiting worker.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_INSTALLED,
+  /// The service worker is activating.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_ACTIVATING,
+  /// The service worker is activated. The service worker in this state is considered an active worker ready to handle functional events.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_ACTIVATED,
+  /// The service worker is redundant.
+  COREWEBVIEW2_SERVICE_WORKER_STATE_REDUNDANT,
+} COREWEBVIEW2_SERVICE_WORKER_STATE;
+
+[uuid(29b994e5-0ac8-5430-89fa-5b4bb2091d8d), object, pointer_default(unique)]
+interface ICoreWebView225 : IUnknown {
+  /// Subscribe to this event that gets raised when a new dedicated worker is created.
+  ///
+  /// A Dedicated Worker is a type of web worker that allow you to run Javascript code in the background without blocking the main thread, making them useful for tasks like heavy computations, data processing, and parallel execution.
+  /// It is "dedicated" because it is linked to a single parent document and cannot be shared with other scripts.
+  ///
+  /// This event is raised when a web application creates a dedicated worker using the
+  /// `new Worker("/worker.js")` method. See the
+  /// [Worker](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker)
+  /// for more information.
```

Comment:
I like this ref doc describing how this works including simple `new Worker(...` sample. Great!